server/api/db/sqldb/db.py

```diff
@@ -897,6 +884,74 @@ def delete_tag_from_artifacts(
             )
         self._delete_artifacts_tags(session, project, artifacts, tags=[tag])
 
+    def tag_artifacts(
+        self,
+        session,
+        tag_name: str,
+        artifacts,
+        project: str,
+    ):
+        objects = []
+        for artifact in artifacts:
+            # remove the tags that point to artifacts with the same key
+            # and a different producer id
+            query = (
+                self._query(
+                    session,
+                    artifact.Tag,
+                    name=tag_name,
+                    project=project,
+                )
+                .join(
+                    ArtifactV2,
+                )
+                .filter(
+                    ArtifactV2.key == artifact.key,
+                    ArtifactV2.project == project,
```

Comment:
Being extra-picky - these predicates are redundant (the query on the tag covers those already)